w06-ruby-on-rails/d4-dawn-intro-rails/README.md

```diff
@@ -0,0 +1,195 @@
+
+## Running Rails (5m)
+
+- Create a new Rails app with `rails new railsFun --skip-activerecord`
+    - the last flag tells the application to not use `activerecord`
+    - `activerecord` is our ORM that manages our Models and our database, we'll learn more about it this afternoon
+- `cd` into your `railsFun` folder and run
+- run `rails server` or just `rails s` and see what happens
+- This will start a server on `localhost:3000` head there and see what it says...
+
+## Rails File Structure (10m)
+
+![Rails File Structure](http://i.imgur.com/whOL4DQ.png)
+
+## Routing (10m)
+
+In `routes.rb` we write logic to map our paths to controllers we will make. See the image below for how the controller actions match up with their respective url paths and HTTP verbs (which combine to define a route).
+
+![Rails Routing](./restful-routing-rails.png)
+
+Let's say when a user sends a `GET` request to the root route, `/`, we want the `WelcomeController`'s `index` method to run. In order to do that we could write:
+
+
+`route.rb`
+
+```ruby
+get "/" => "welcome#index"`
+```
+Try saving the file and heading to your root route. What error do you get?
+
+**Pro tip:** run `rake routes` in your rails app to see a list of all the application's routes. The output will be similar to the image above.
+
+## Controllers (10m)
+
+If you did the last step correctly, you should see an error message: `*uninitialized constant WelcomeController*`. This means that we need to create a controller with the name `welcome` as that is where we told our route to go in the first place!
+
+- Run: `rails generate controller welcome`
+
+This will generate a file in the Controllers directory called `welcome_controller.rb`, along with several other directories relating to it, including views, javascripts, stylesheets.
+
+Reload the page again and find a different error message: `*The action 'index' could not be found for WelcomeController*`. We have created the welcome controller correctly, but there is no `index` method defined. Let's make one:
+
+`welcome_controller.rb`
+
+```ruby
+class WelcomeController < ApplicationController
+    def index
+    end
+end
+```
+
+Wow another error! `*Missing template welcome/index...*` Since we have a `welcome` controller and an `index` method, Rails automatically will try to render a view with the path `app/views/welcome/index.html.erb`.  A directory `/welcome` should already exist, it was generated when the welcome controller was generated (thanks, Rails!). Inside of there add the file `index.html.erb` and inside of the file add some html:
+
+app/views/welcome/index.html.erb
+
+```html
+<h1>I make internets with Rails</h1>
+<img src="http://i.giphy.com/SPZFhfUJjsJO0.gif" alt="learning internet" style="width: 300px">
+```
+Visit your root route one more time.
+
+Nice!
+
+## View (5m)
+
+- This will not change the behavior, but you may wish to be more explicit in our controller by stating to `render` the template `index`
```

Comment:
we may wish to be more explicit in our controller 
or
you may wish to be more explicit in your controller